src/app/componenets/launch-page/launch-page.component.ts

```diff
@@ -0,0 +1,207 @@
+    );
+  }
+
+  public imagesToArray(images: []): void {
+    this.images = images.map((image) => ({
+      name: image,
+    }));
+  }
+
+  public timelineStepperSetter(
+    timeline: ITimeline[],
+    startFrom: number,
+    currentSeconds: number,
+    stepper: MatStepper,
+    backwards?: boolean
+  ): void {
+    let seconds = backwards ? currentSeconds : startFrom - currentSeconds;
+    timeline.forEach((element, index) => {
+      if (seconds == element.seconds) {
+        element.isVisible = true;
+        index + this.offset > timeline.length ? timeline[index + this.offset].isVisible = true : null;
+          setTimeout(() => {
+            index - this.offset >= 0 ? timeline[index - this.offset].isVisible = false : null;
+            stepper.selectedIndex = this.offset;
+            this.cdref.detectChanges();
+          }, 90);
+          timeline[index + this.offset + this.offset] !== undefined ? timeline[index + this.offset + this.offset].isVisible = true : null;
+      }
+    });
+  }
+
+  private setVisible(element:ITimeline[], index:number): void {
+    element[index].isVisible = true;
+  }
+
+  private startAirTimer(end: number): void {
+    this.airTimeline[0].isVisible = true;
+    this.airTimeline[1].isVisible = true;
+    const sub$ = interval(this.timeInterval).subscribe((sec) => {
+      this.airProgressbarValue.next(100 - (sec * 100) / end);
+      this.secInAir.next(sec);
+
+      this.timelineStepperSetter(this.airTimeline, end, this.secInAir.getValue(), this.stepperAir, true);
+
+      if (this.secInAir.getValue() === end) {
+        sub$.unsubscribe();
+      }
+    });
+  }
+
+  private startGroundTimer(begin: number, end: number): void {
+    const sub$ = interval(this.timeInterval).subscribe((sec) => {
+      this.groundProgressbarValue.next(100 - (sec * 100) / begin);
+      this.secTillLiftoff.next(sec);
+
+      this.timelineStepperSetter(this.groundTimeline, begin, this.secTillLiftoff.getValue(), this.stepperGround);
+
+      if (this.secTillLiftoff.getValue() === begin) {
+        sub$.unsubscribe();
+        this.groundTimeline = this.groundTimeline.map((event) => ({
+          ...event,
+          isVisible: false,
+        }));
+        this.startAirTimer(end);
+      }
+    });
+  }
+
+  public capitalizeHeader(event: string): string {
+    let header = event.replace(/_/g, ' ');
+    return header[0].toUpperCase() + header.slice(1).toLowerCase();
+  }
+
+  public buildTimeline(timeline: {}): void {
+    let timelineArray = Object.entries(timeline).map(([event, time]) => ({ event, time }));
+    // @ts-ignore
+    timelineArray.sort((a, b) => a.time - b.time);
+
+    this.groundTimeline = timelineArray
+      .filter((element) => (element.time as number) < 0)
+      .map((item) => ({
+        seconds: (item.time as number) * -1,
+        header: this.capitalizeHeader(item.event),
+        isVisible: false,
+        more: '~ ' + (item.time as number) * -1 + ' s. till liftoff',
```

Comment:
To już najlepiej w templatce ustawiać. W plikach TS przekazuj same wartości. Prezentacją niech zajmuje się HTML

Reply:
Też o tym myślałem. Jednak jest tam mnożenie przez `-1` co jednak jest to operacją matematyczną.

Reply:
no tak, dlatego możesz ustawić:
`time: item.time * -1`
a sama prezentacja może być już zrobiona w html

Reply:
True